tests/core/pthread/create.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
```

Comment:
we have tests/pthread/test_pthread_create.cpp already. How does tests/core/pthread relate to the tests in tests/pthread?

Reply:
`tests/pthread/` are existing tests, that only work on the web. The new `core/pthreads` are for the core test suite and run on node.

Sadly the web tests can't just be reused as they make some web assumptions (like reporting the result using an xhr, when it's ok to block and when not, etc., which it turns out node differs on).

Reply:
I see. We should probably document somewhere (even if it's just for our own testing purposes) exactly the differences between web-pthreads and node-pthreads environments, and it would be nice to eventually unify the tests. But that doesn't need to be in this PR.